File: packages/typescript/src/roles/technical-writer.ts
import { Agent } from "../agent.js";
import { AgentRole, TaskType } from "../types.js";
import type { AgentConfig } from "../types.js";

const HANDLED_TASK_TYPES: TaskType[] = [
  TaskType.Documentation,
];

export class TechnicalWriter extends Agent {
  constructor(config: Partial<AgentConfig> = {}) {
    super({
      role: AgentRole.TechnicalWriter,
      ...config,
    });
  }

  canHandle(taskType: TaskType): boolean {
    return HANDLED_TASK_TYPES.includes(taskType);
  }
}

File: packages/python/src/tcs_bpi_ai_team/roles/technical_writer.py
"""Technical Writer agent."""

from __future__ import annotations

from tcs_bpi_ai_team.agent import Agent
from tcs_bpi_ai_team.types import AgentConfig, AgentRole, TaskType

_HANDLED_TASK_TYPES = [
    TaskType.DOCUMENTATION,
    TaskType.BACKEND_FEATURE,
    TaskType.FRONTEND_FEATURE,
]


class TechnicalWriter(Agent):
    """Technical Writer — documentation, API docs, runbooks."""

    def __init__(self, config: AgentConfig | None = None) -> None:
        super().__init__(config or AgentConfig(role=AgentRole.TECHNICAL_WRITER))

    def can_handle(self, task_type: TaskType) -> bool:
        return task_type in _HANDLED_TASK_TYPES
